### ISSUE TITLE
Image_advanced only return first image id 

### ISSUE THREAD

**Issue by Giacomo92** [wpmetabox/meta-box#1235]
I have the last version of metabox plugin and I follow the doc to create a image_advanced field.
Here the code for register the metabox:
 
   
```
array(
          'id' => 'gallery-images',
	  'name' => __('gallery-images','stm_vehicles_listing'),
	  'type' => 'image_advanced',
           'force_delete' => false,
           'max_file_uploads' => 10,
           'max_status' => 'false',
	   'image_size' => 'thumbnail',
	   'tab' => 'photosandvideos',
),
```

I register the metabox with:
`add_filter( 'rwmb_meta_boxes', 'prefix_register_meta_boxes' );`           

When I try to get the value on frontend.i get only ID of the first image. I have already tried all this alternative from the referenced isuees:
```
$post_id = get_the_ID(); // int(1264) --> The post ID is not empty

$gallery = rwmb_meta('gallery-images'); //string(4) "1530" -> The id of the first image 
$gallery = rwmb_meta('gallery-images','size=full',$post_id); //string(4) "1530" -> The id of the first image 
$gallery = rwmb_meta('gallery-images','size=full'); //string(4) "1530" -> The id of the first image 
$gallery = rwmb_meta('gallery-images','size=full',$post_id); //string(4) "1530" -> The id of the first image 
$gallery = rwmb_meta('gallery-images','size=thumbnail'); //string(4) "1530" -> The id of the first image 
$gallery = rwmb_meta('gallery-images'); //string(4) "1530" -> The id of the first image 
$gallery = rwmb_meta('gallery-images',array( 'size' => 'full' ),$post_id); //string(4) "1530" -> The id of the first image 
$gallery = rwmb_meta('gallery-images',array( 'size' => 'full' )); //string(4) "1530" -> The id of the first image 
$gallery = rwmb_meta('gallery-images',array( 'size' => 'thumbnail' ),$post_id); //string(4) "1530" -> The id of the first image 
$gallery = rwmb_meta('gallery-images',array( 'size' => 'thumbnail' )); //string(4) "1530" -> The id of the first image 

```

On the backend I set and I can see my three image. Why i can get an array of image from $gallery?

**Comment by rilwis** [comment]
Hello, I've just tried your code and it returns the full array of images. See this screenshot:

https://i.imgur.com/WhJNxTL.png

Here is my test code:

```php
<?php
add_filter( 'rwmb_meta_boxes', function ( $meta_boxes ) {
	$meta_boxes[] = [
		'title'  => 'Test',
		'fields' => [
			array(
				'id'               => 'gallery-images',
				'name'             => __( 'gallery-images', 'stm_vehicles_listing' ),
				'type'             => 'image_advanced',
				'force_delete'     => false,
				'max_file_uploads' => 10,
				'max_status'       => 'false',
				'image_size'       => 'thumbnail',
				'tab'              => 'photosandvideos',
			),
		],
	];
	return $meta_boxes;
} );

add_filter( 'the_content', function( $content ) {
	if ( is_single() ) {
		$gallery = rwmb_meta('gallery-images');
		$content .= '<pre>' . print_r( $gallery, true ) . '</pre>';
	}
	return $content;
} );
```

Please check whether you `add_filter` only for the admin area or not. If that's the case, then try:

```php
$gallery = rwmb_meta( 'gallery-images', ['type' => 'image_advanced'] );
```

But it's not recommended to add the filter only in the admin.

**Comment by Giacomo92** [comment]
Thank you so much @rilwis. I was at my wits' end :-)
This `$gallery = rwmb_meta( 'gallery-images', ['type' => 'image_advanced'] );` fix this issue and also the [#1234](https://github.com/wpmetabox/meta-box/issues/1234) one. I have add my code on a 3rd part plugin that restrict `add_filter` only for the admin area like you said.

**Comment by rilwis** [comment]
Glad that it solves the problem!
